hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -518,6 +485,27 @@ private ContainerMoveSelection matchSourceWithTarget(
       }
       return null;
     }
+
+    //if the utilization of the source data node becomes lower than lowerLimit
+    //after the container is moved out , then the container can not be
+    // a candidate one, and we should remove it from the candidateContainers.
+    candidateContainers.removeIf(c -> {
+      ContainerInfo cInfo;
+      try {
+        cInfo = containerManager.getContainer(c);
+      } catch (ContainerNotFoundException e) {
+        LOG.warn("Could not find container {} when " +
+            "be matched with a move target", c);
+        //remove this not found container
+        return true;
+      }
+      Long totalLeavingSize = sizeLeavingNode.get(source) +
+          cInfo.getUsedBytes();
+      return Double.compare(nodeManager.getUsageInfo(source)
+        .calculateUtilization(totalLeavingSize), lowerLimit) < 0 ||
+          totalLeavingSize > config.getMaxSizeLeavingSource();
+    });
```

Comment:
This block of code is analogous to what `canSizeEnterTarget` does for target nodes. Can we have a similar method called `canSizeLeaveSource` for source nodes?
Also, the `calculateUtilization(long plusSize)` method adds the specified plus size to used space. Here, I think we want to subtract `totalLeavingSize` from used space, instead.

Reply:
thanks for pointing out this,  will fix it!